src/js/containers/search/helpers/searchAnalytics.js

```diff
@@ -0,0 +1,235 @@
+import { recipientTypes, groupLabels } from 'dataMapping/search/recipientType';
+import {
+    pricingTypeDefinitions,
+    setAsideDefinitions,
+    extentCompetedDefinitions
+} from 'dataMapping/search/contractFields';
+
+import Analytics from 'helpers/analytics/Analytics';
+
+const eventCategory = 'Advanced Search - Search Filter';
+
+export const convertDateRange = (range) => {
+    if (range.length !== 2) {
+        // this must be an array of length 2
+        return null;
+    }
+    else if (!range[0] && !range[1]) {
+        // no start or end dates are set
+        return null;
+    }
+
+    const startDate = range[0] || '...';
+    const endDate = range[1] || 'present';
+
+    return [{
+        action: 'Time Period - Date Range',
+        label: `${startDate} to ${endDate}`
+    }];
+};
+
+export const parseAgency = (agency) => {
+    const toptier = agency.toptier_agency;
+    const subtier = agency.subtier_agency;
+    const office = agency.office_agency;
+    if (agency.agencyType === 'toptier') {
+        if (toptier.abbreviation) {
+            return `${toptier.name} (${toptier.abbreviation})/${toptier.cgac_code}`;
+        }
+        return `${toptier.name}/${toptier.cgac_code}`;
+    }
+    else if (agency.agencyType === 'subtier') {
+        if (subtier.abbreviation) {
+            return `${subtier.name} (${subtier.abbreviation})/${subtier.subtier_code} - ${toptier.name}/${toptier.cgac_code}`;
+        }
+        return `${subtier.name}/${subtier.subtier_code} - ${toptier.name}/${toptier.cgac_code}`;
+    }
+    else if (agency.agencyType === 'office') {
+        return `${office.name}/${office.aac_code} - ${toptier.name}/${toptier.cgac_code}`;
+    }
+    return null;
+};
+
+
+export const convertReducibleValue = (value, type, parser) => (
+    value.reduce((events, item) => {
+        events.push({
+            action: type,
+            label: (parser && parser(item)) || item
+        });
+        return events;
+    }, [])
+);
+
+export const convertTimePeriod = (value) => {
+    if (Set.isSet(value)) {
+        return convertReducibleValue(
+            value,
+            'Time Period - Fiscal Year'
+        );
+    }
+    else if (Array.isArray(value)) {
+        return convertDateRange(value);
+    }
+    return null;
+};
+
+export const convertAgency = (agencies, type) => (
+    convertReducibleValue(
+        agencies,
+        type,
+        parseAgency
+    )
+);
+
+
+export const convertLocation = (locations, type) => (
+    convertReducibleValue(
+        locations,
+        type,
+        (location) => `${location.display.entity} - ${location.display.standalone}`
+    )
+);
+
+export const convertFilter = (type, value) => {
+    switch (type) {
+        case 'timePeriod':
+            return convertTimePeriod(value);
+        case 'awardType':
+            return convertReducibleValue(
+                value,
+                'Award Type',
+                (item) => awardTypeCodes[item] || item
```

Comment:
Would we want to use a group label (similar to recipient type) to capture the parent checkbox for all contracts/grants/etc.?